docs/src/mock.md

```diff
@@ -3,121 +3,524 @@ id: mock
-});
+    await route.FulfillAsync(new RouteFulfillResponse
+    {
+      Response = response,
+      Json = json
+    });
+  }
+);
+// Go to the page
+await page.GotoAsync("https://demo.playwright.dev/api-mocking");
+
+// Assert that the Strawberry fruit is visible
+await Expect(page.GetByTextAsync("Playwright", new() { Exact: true })).ToBeVisibleAsync();
 ```
 
 ```java
-page.route("https://dog.ceo/api/breeds/list/all", route -> {
-  APIResponse response = route.fetch();
-  JsonObject json = new Gson().fromJson(response.text(), JsonObject.class);
-  JsonObject message = json.get("message").getAsJsonObject();
-  message.set("big_red_dog", new JsonArray());
+page.route("*/**/api/v1/fruits", route -> {
+  Response response = route.fetch();
+  Map<String, Object> json = response.json();
+  json.add(new HashMap<String, Object> { { "name", "Playwright" }, { "id", 100 } });
   // Fulfill using the original response, while patching the response body
   // with the given JSON object.
-  route.fulfill(new Route.FulfillOptions()
-    .setResponse(response)
-    .setBody(json.toString()));
+  route.fulfill(new Route.FulfillOptions().setResponse(response).setJsonBody(json));
+});
+
+// Go to the page
+page.goto("https://demo.playwright.dev/api-mocking");
+
+// Assert that the Strawberry fruit is visible
+assertThat(page.getByText("Playwright", new Page.GetByTextOptions().setExact(true))).isVisible();
+```
+From the trace of our test we can see that the API was called and the response was modified.
+![trace of test showing api being called and fulfilled](https://github.com/microsoft/playwright/assets/13063165/8b8dd82d-1b3e-428e-871b-840581fed439)
+
+By inspecting the response we can see that our new fruit was added to the list.
+![trace of test showing the mock response](https://github.com/microsoft/playwright/assets/13063165/03e6c87c-4ecc-47e8-9ca0-30fface25e9d)
+
+Read more about [advanced networking](./network.md).
+
+## Mocking with HAR files
+
+A HAR file is an HTTP Archive file that contains a record of all the network requests that are made when a page is loaded. It contains information about the request and response headers, cookies, content, timings, and more. You can use HAR files to mock network requests in your tests. You'll need to:
+
+1. Record a HAR file.
+1. Commit the HAR file alongside the tests.
+1. Route requests using the saved HAR files in the tests.
+
+### Recording a HAR file
+
+To record a HAR file we use the `routeFromHAR` method. This method takes in the path to the HAR file and an optional object of options.
+The options object can contain the URL so that only requests with the URL matching the specified glob pattern will be served from the HAR File. If not specified, all requests will be served from the HAR file.
+
+The `update` option updates the given HAR file with the actual network information or creates a new HAR file, in the hars folder, if one doesn't already exist. In order to record the HAR file, set `update` to true.
```

Comment:
```suggestion
Setting `update` option to true will create or update the HAR file with the actual network information instead of serving the requests from the HAR file. Use it when creating a test to populate the HAR with real data.
```